velox/common/file/tests/FaultyFile.cpp

```diff
@@ -54,14 +54,24 @@ uint64_t FaultyReadFile::preadv(
 }
 
 FaultyWriteFile::FaultyWriteFile(
+    const std::string& path,
     std::shared_ptr<WriteFile> delegatedFile,
     FileFaultInjectionHook injectionHook)
-    : delegatedFile_(std::move(delegatedFile)),
+    : path_(path),
+      delegatedFile_(std::move(delegatedFile)),
       injectionHook_(std::move(injectionHook)) {
   VELOX_CHECK_NOT_NULL(delegatedFile_);
 }
 
 void FaultyWriteFile::append(std::string_view data) {
+  if (injectionHook_ != nullptr) {
+    FaultFileWriteOperation op(path_, data);
+    injectionHook_(&op);
+    if (op.delegate) {
```

Comment:
```
if (!op.delegate) {
   return;
}
```

Reply:
Then I have to write it as:
```
if (!op.delegate) {
      return;
    }
    else {
      delegatedFile_->append(op.data);
      return;
    }
```

because I should only call `delegatedFile_->append(op.data);` once

Reply:
I mean
```
if (injectionHook_ != nullptr) {
    FaultFileWriteOperation op(path_, data);
    injectionHook_(&op);
    if (!op.delegate) {
       return;
    }
}
delegatedFile_->append(op.data);
```

Reply:
`op` can not be used out of the `if` scope

Reply:
delegatedFile_->append(data);

You can just pass data?